superset/dataframe.py

```diff
@@ -119,6 +119,8 @@ def __init__(self, data, cursor_description, db_engine_spec):
             }
             self.df = pd.DataFrame(data, columns=column_names)
         else:
+            if getattr(db_engine_spec, "odbc_row_results", None):
+                data = [[value for value in row] for row in data]
```

Comment:
Can you explain why this is needed? Also, if the attribute is added to `BaseEngineSpec`, you should not need to test for it first (see below).

Reply:
The problem with fixing this behaviour per engine is that several engines support both odbc and non-odbc drivers (e.g. mssql referenced in the test). Does the default behaviour not work for exa? As pyodbc has been supported for quite some time on other engines, it feels strange that this should need to be handled differently in this case.

Reply:
The default behaviour fails during `pd.Dataframe` instantiation with messages similar to `Shape of passed values is (1, 1), indices imply (1, 50)`, or when I `SELECT` only one column (where the two cases converge and the dataframe can be created) it fails at a later stage with `Unserializable [<class 'pyodbc.Row'>]`

While debugging I also tried to leave the data list as is and pass column names like `[["col1"], ["col2"], ["col3"]]` as per the test, but this was not successful either. 

> As pyodbc has been supported for quite some time on other engines, it feels strange that this should need to be handled differently in this case.

🤷‍♂

Reply:
Would it be preferable to add a try/expect block instead?

Reply:
That still feels hacky, so would prefer not to go down that route. My intuition tells me there might be a bug in the SqlAlchemy connector. Ideally we would like to end up with a solution that works identically for all odbc sources.

Reply:
If something special is needed for Exasol/odbc it should be in the engine specific module. One way would be to override `fetch_data` for Exasol.
https://github.com/apache/incubator-superset/blob/master/superset/db_engine_specs/base.py#L225